### PR TITLE
Test for RemoveZoneAndAssociatedRecords

### PR DESCRIPTION
**Pull request by DongLieu**
## 1. Summary
Fixes #617

## 2.Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## 5. Checklist

- [ ] Does the Readme need to be updated?

### CODE REVIEW

**Comment by joe-bowman** [comment]
Can we also check for validators / receipts / withdrawal records etc. by zone to ensure that everything that should have been cleaned up has been? thankjs,

**Comment by DongLieu** [comment]
Cannot check:

- delegation records: because the zone has been deleted

- performance delegation records: because the zone has been deleted

- validators: In the [RemoveZoneAndAssociatedRecords](https://github.com/quicksilver-zone/quicksilver/blob/main/x/interchainstaking/keeper/zones.go#L436) function do not remove Validator. Set add validators to create related

Additional:
In [GetReceipt](https://github.com/quicksilver-zone/quicksilver/blob/main/x/interchainstaking/keeper/receipt.go#L320), I think we should pass by value: 'chainID' and 'TxHash' instead of the current 'key'

**Comment by joe-bowman** [comment]
Validators should definitely be removed if a zone is removed. Can you make a fix for this, please? 

Validator storage changed in v1.4 and the deletion of validators for a given zone was almost certainly overlooked.

**Comment by joe-bowman** [comment]
> * delegation records: because the zone has been deleted
> * performance delegation records: because the zone has been deleted

Also probably a good time to refactor:

k.IterateAllDelegations(ctx, zone, func(delegation types.Delegation) (stop bool) {
k.IterateAllPerformanceDelegations(ctx, zone, func(delegation types.Delegation) (stop bool) {
k.IterateZoneReceipts(ctx, zone, func(index int64, receiptInfo types.Receipt) (stop bool) {
				
to take chainID instead of zone (all three methods only make use of zone.ChainID internally anyway!) but allows us to assert the removal properly.

**Comment by DongLieu** [comment]
Okay,
Do I create a new PR to do this, or do it in this PR?

**Comment by DongLieu** [comment]
For checks delegation and performance delegation records, I need to wait for the pull #748 to be merged

**Comment by DongLieu** [comment]
I need to merge this PR #758 first, please review it

**Comment by DongLieu** [comment]
Ok, done. @ThanhNhann @joe-bowman

File: x/interchainstaking/keeper/delegation_test.go [hunk 1]
```diff
@@ -634,7 +634,7 @@ func (suite *KeeperTestSuite) TestFlushOutstandingDelegations() {
 func (suite *KeeperTestSuite) TestPerformanceDelegation() {
 	quicksilver := suite.GetQuicksilverApp(suite.chainA)
 	ctx := suite.chainA.GetContext()
-	chainID := "quicksilver-1"
```
Comment (ThanhNhann) [inline x/interchainstaking/keeper/delegation_test.go:637]: Why we should use `const` here?

**Review by anhductn2001 — APPROVED** [review]
Nice works @DongLieu !

**Review by joe-bowman — APPROVED** [review]
lgtm

**Pull request merged**